src/index.js

```diff
@@ -82,23 +87,37 @@ async function run() {
       // skipVersionFile can mean there is no version file and skipCommit can mean that the user
       // is only interested in tags
       if (skipVersionFile || skipCommit) {
-        core.info('Using GIT to determine the new version');
-        const versioning = await handleVersioningByExtension('git', versionFile, versionPath, recommendation.releaseType)
+        core.info('Using GIT to determine the new version')
+        const versioning = await handleVersioningByExtension(
+          'git',
+          versionFile,
+          versionPath,
+          recommendation.releaseType
+        )
         newVersion = versioning.newVersion
-
       } else {
-        const files = versionFile.split(',').map(f => f.trim())
+        const files = versionFile.split(',').map((f) => f.trim())
         core.info(`Files to bump: ${files.join(', ')}`)
 
-        const versioning = await Promise.all(files.map((file) => {
-          const fileExtension = file.split('.').pop()
-          core.info(`Bumping version to file "${file}" with extension "${fileExtension}"`)
-          return handleVersioningByExtension(fileExtension, file, versionPath, recommendation.releaseType)
-        }));
+        const versioning = await Promise.all(
+          files.map((file) => {
+            const fileExtension = file.split('.').pop()
+            core.info(`Bumping version to file "${file}" with extension "${fileExtension}"`)
+            return handleVersioningByExtension(fileExtension, file, versionPath, recommendation.releaseType)
+          })
+        )
 
         newVersion = versioning[0].newVersion
       }
 
+      const gitTag = `${tagPrefix}${newVersion}`
+
+      if (preChangelogGeneration) {
+        newVersion = await require(path.resolve(preChangelogGeneration)).preChangelogGeneration({
```

Comment:
Maybe an idea to return `version` and `tag` so you can define both? As now you are only able to change the version and not the tag.

Reply:
I thought that since currently tag is always equal to `prefix + version`, there was no real need to redefine the tag. But it could be useful in some cases, so I will add it